### PR TITLE
Add SummarizableMetricAggregation base class

### PR DESCRIPTION
**Pull request by dupuyjs**
The following changes are proposed:

Add SummarizableMetricAggregation base class. This class defines if a metric aggregation can be used with summarize operator. Applied on Average, Cardinality, ExtendStats, Max, Min, Percentile and Sum.

### CODE REVIEW

**Comment by AsafMah** [comment]
So this interface's only goal is to mark which metrics can be summerizable, which right now are everyone except top hits?

**Review by eladiw — APPROVED** [review]
LGTM I would add a section in the docs to document all of this knowledge

**Comment by dupuyjs** [comment]
> So this interface's only goal is to mark which metrics can be summerizable, which right now are everyone except top hits?

Yes, this is the exact and only purpose, ISummarizable interface identifies metrics where summarize can be applied. TopHits will be the only one with another interface (IPartitionable) at the moment.

**Comment by AsafMah** [comment]
I wrote it in the other one too, I'm not a big fan of marker interfaces, and here especially because you only mark MetricAggregations.
Is there a reason we couldn't have it as part of the hierarchy (SummarizableMetricAggragegation inherits from MetricAggregation)?

**Comment by dupuyjs** [comment]
> I wrote it in the other one too, I'm not a big fan of marker interfaces, and here especially because you only mark MetricAggregations. Is there a reason we couldn't have it as part of the hierarchy (SummarizableMetricAggragegation inherits from MetricAggregation)?

No objection. Let me check.

File: K2Bridge/Visitors/Aggregations/AggregationContainerVisitor.cs [hunk 1]
```diff
@@ -36,13 +37,17 @@ public string BuildSummarizableMetricsQuery(AggregationDictionary aggregationDic
         {
             var query = new StringBuilder();
 
-            // Collect all metrics
+            // Collect all ISummarizable metrics
```
Comment (AsafMah) [inline K2Bridge/Visitors/Aggregations/AggregationContainerVisitor.cs:40]: It's no longer ISummarizable.

File: K2Bridge/Visitors/Aggregations/AggregationContainerVisitor.cs [hunk 2]
```diff
@@ -36,13 +37,17 @@ public string BuildSummarizableMetricsQuery(AggregationDictionary aggregationDic
...
             // ['2']=max(AvgTicketPrice), ['3']=avg(DistanceKilometers)
             var summarizableMetrics = new List<string>();
-            foreach (var (_, aggregation) in aggregationDictionary)
+            foreach (var (_, aggregationContainer) in aggregationDictionary)
```
Comment (AsafMah) [inline K2Bridge/Visitors/Aggregations/AggregationContainerVisitor.cs:43]: Get the revised method from 3/3

File: K2Bridge/Visitors/Aggregations/AggregationContainerVisitor.cs (hunk 1 above)
Comment (dupuyjs) [reply K2Bridge/Visitors/Aggregations/AggregationContainerVisitor.cs]: Fixed.

File: K2Bridge/Visitors/Aggregations/AggregationContainerVisitor.cs (hunk 2 above)
Comment (dupuyjs) [reply K2Bridge/Visitors/Aggregations/AggregationContainerVisitor.cs]: Done.

**Review by AsafMah — APPROVED** [review]

**Pull request merged**